src/sisl/geom/flat.py

```diff
@@ -73,3 +73,112 @@ def graphene(bond=1.42, atoms=None, orthogonal=False):
+    """
+
+    # Function that generates one of the six triangular portions of the
+    # hexagonal flake. The rest of the portions are obtained by rotating
+    # this one by 60, 120, 180, 240 and 300 degrees.
+    def _minimal_op(shells):
+
+        # The function is based on the horizontal lines of the hexagon,
+        # which are made of a pair of atoms.
+        # For each shell, we first need to complete the incomplete horizontal
+        # lines of the previous shell, and then branch them up and down to create
+        # the next horizontal lines.
+
+        # Displacement from the end of one horizontal pair to the beggining of the next
+        branch_displ_x = bond * 0.5 # cos(60) = 0.5
+        branch_displ_y = bond * 3 ** 0.5 / 2 # sin(60) = sqrt(3)/2
+
+        # Iterate over shells. We also keep track of the atom types, in case
+        # we have two different atoms in the honeycomb lattice.
+        op = np.array([[bond, 0, 0]])
+        types = np.array([0])
+        for shell in range(shells):
+            n_new_branches = 2 + shell
+            prev_edge = branch_displ_y * (shell)
+            
+            sat = np.zeros((shell + 1, 3))
+            sat[:, 0] = op[-1, 0] + bond
+            sat[:, 1] = np.linspace(-prev_edge, prev_edge, shell + 1)
+            
+            edge = branch_displ_y * (shell + 1)
+
+            branches = np.zeros((n_new_branches, 3))
+            branches[:, 0] = sat[0, 0] + branch_displ_x
+            branches[:, 1] = np.linspace(-edge, edge, n_new_branches )
+
+            op = np.concatenate([op, sat, branches])
+            types = np.concatenate([types, np.full(len(sat), 1), np.full(len(branches), 0)])
+
+        return op, types
+    
+    # Get the coordinates of 1/6 of the hexagon for the requested number of shells.
+    op, types = _minimal_op(shells)
+
+    single_atom_type = isinstance(atoms, (str, Atom)) or len(atoms) == 1
+
+    # Create a geometry from the coordinates.
+    ats = atoms  if single_atom_type else np.asarray(atoms)[types]
+    geom = Geometry(op, atoms=ats)
+
+    # The second portion of the hexagon is obtained by rotating the first one by 60 degrees.
+    # However, if there are two different atoms in the honeycomb lattice, we need to reverse the types.
+    next_triangle = geom if single_atom_type else Geometry(op, atoms=np.asarray(atoms)[types - 1])
+    geom += next_triangle.rotate(60, [0,0,1])
+
+    # Then just rotate the two triangles by 120 and 240 degrees to get the full hexagon.
+    geom += geom.rotate(120, [0,0,1]) + geom.rotate(240, [0,0,1])
+
+    # Set the cell according to the requested vacuum
+    max_x = np.max(geom.xyz[:,0])
+    geom.cell[0,0] = max_x * 2 + vacuum
+    geom.cell[1,1] = max_x * 2 + vacuum
+    geom.cell[2,2] = 20.
+
+    # Center the flake
+    geom = geom.translate(geom.center(what="cell"))
+
+    # Set boundary conditions
+    geom.lattice.set_boundary_condition(a="Unknown", b="Unknown", c="Unknown")
```

Comment:
No, call `geometry_set_nsc(geom, [False, False, False])`, `geometry_set_nsc` does other things as well. :)

Reply:
The only thing that it does is to set nsc to `[1, 1, 1]`, no? But to do that it calls first `geometry.optimize_nsc` which is a waste of time in this case...

Reply:
The point is that it we have a single reference function that sets up everything. All `geom` codes call this one. And so it should also.

Reply:
Hmm ok but then I think `geometry_set_nsc` should check if all directions are not periodic and don't call `optimize_nsc` then

Reply:
True, I'll do that afterwards, call it and I'll get it in.

Reply:
Done